### ISSUE TITLE
How to set this up in a Node.JS app?

### ISSUE THREAD

**Issue by xblox** [dojo/loader#106]
Hi,
sorry for bothering but do you have by chance some instructions how to set this up in a Node.JS application? I tried to get them from the test units but I failed in doing so. 

I did:
`npm install --save @dojo/loader `

and then: 

`import * as loader from '@dojo/loader'; ` 

this makes indeed require/define visible but I can't any type completion in VSCode nor I don't know how to pass a configuration for the loader. I tried then extending 'files' in my tsconfig for "./node_modules/@dojo/loader/interfaces.d.ts"  but that seems odd (don't need this for other TS packages) and still doesn't give me any interfaces.

Basically I want the obvious (requiring my dojo-1 AMD server compatible modules and mix them with new TS classes): 

```ts
import {Config, Loader} from '@dojo/loader';

const config:Config = {packages:[{name:'foo', location:'absolutePathToFoo' }]}
const loader:Loader = Loader(Config); // or whatever it needs to get the AMD-Require

const myAMDModule = loader(['foo/module'],function(fooModule){
   ....
});
```
I can post this on SO if you prefer. Close this here just if its out of scope for you, for me it's essential as Dojo-1 user :-)

Thanks a lot already.

**Comment by dylans** [comment]
I'm re-opening this, as this is something we need to cover in our migration guide.

I'll see if we have an example we can share shortly.

**Comment by rorticus** [comment]
There is no great way to do this right now.. I've [opened an issue](https://github.com/dojo/loader/issues/108) to make this easier to use in NodeJS.

For now, you can use the interfaces defined in [@dojo/interfaces](https://github.com/dojo/interfaces/blob/master/src/loader.d.ts) to help load the loader with typings.

It looks something like,

```ts
import { RootRequire } from '@dojo/interfaces/loader';
const dojoRequire: RootRequire = require('@dojo/loader');
```

You can see a simple example here,

https://github.com/rorticus/node-loader-example

Hope this helps!

**Comment by xblox** [comment]
@rorticus & @dylans  :  fantastic! thanks a lot for the heads up!

**Comment by dylans** [comment]
Closing this question in favor of the newly opened issue.
